deepchem/splits/splitters.py

```diff
@@ -1360,8 +1410,6 @@ class ScaffoldSplitter(Splitter):
   ... dataset = dc.data.DiskDataset.from_numpy(X=Xs,y=Ys,w=np.zeros(len(data_test)),ids=data_test)
   >>> scaffoldsplitter = dc.splits.ScaffoldSplitter()
   >>> train,test = scaffoldsplitter.train_test_split(dataset)
-  >>> train
```

Comment:
`ScaffoldSplitter` behavior is random and depends the  on `seed` parameter. We can't predict the exact content of the `train`, so doctest will fail if we set any outcome in the example. We can fix `seed` in this example, but I am afraid it will be broken in the feature when we update numpy version for instance.

Reply:
See comment above. `ScaffoldSplitter` should be deterministic and tested to be deterministic. That might be a nice fix for a future PR!